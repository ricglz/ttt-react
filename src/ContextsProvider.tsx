import { useTranslation } from "react-i18next";
import { Toaster } from "react-hot-toast";
import { BrowserRouter as Router } from "react-router-dom";
import type { FC, ReactNode } from "react";

type Locale = string;

function getDirection(locale: Locale) {
  return locale === "ar" ? ["rtl", "text-right"] : ["ltr", "text-left"];
}

type Props = {
  ChildrenComponent: FC;
  children?: ReactNode;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  RouterComponent?: any;
};

export default function ContextsProvider({
  children,
  ChildrenComponent,
  RouterComponent = Router,
}: Props) {
  const { i18n } = useTranslation();
  const [direction, klass] = getDirection(i18n.language);

  return (
    <div dir={direction} className={klass}>
      <RouterComponent>
        <ChildrenComponent />
        {children}
      </RouterComponent>
      <Toaster position="bottom-right" />
    </div>
  );
}
