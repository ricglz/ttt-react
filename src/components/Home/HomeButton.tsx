import type { Locale } from 'react-i18next';
import { Link } from 'react-router-dom';
import { FormattedMessage } from '../Layout/FormattedText';

type RenderTextProps =
  | { staticText?: false; text: Locale }
  | { staticText?: true; text: string };

function renderText({ staticText, text }: RenderTextProps) {
  return staticText ? (
    <span>{text}</span>
  ) : (
    <FormattedMessage locale={text as Locale} />
  );
}

type Props = { url: string } & RenderTextProps;

const HomeButton = ({ url, ...rest }: Props) => (
  <div className="row justify-content-center border-top py-3">
    <Link className="btn btn-home" to={url}>
      {renderText(rest)}
    </Link>
  </div>
);

export default HomeButton;
