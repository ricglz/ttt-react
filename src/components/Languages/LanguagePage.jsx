import React from 'react';
import LanguageButton from '../Layout/LanguageButton';
import LANGUAGES from './Languages';

const renderButtons = ({ changeLocale, locale, history }) => LANGUAGES.map((lang) => (
  <LanguageButton
    changeLocale={changeLocale}
    currentLocale={locale}
    history={history}
    locale={lang}
    key={lang}
  />
));

const LanguagePage = (props) => (
  <div className="row justify-content-center mt-5">{renderButtons(props)}</div>
);

export default LanguagePage;
