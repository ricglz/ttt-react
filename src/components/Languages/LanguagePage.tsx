import { useTranslation } from 'react-i18next';
import LanguageButton from './LanguageButton';

export default function LanguagePage() {
  const { i18n } = useTranslation();
  const currentLocale = i18n.language;
  return (
    <div className="row justify-content-center mt-5">
      {i18n.languages.map((lang) => (
        <LanguageButton
          key={lang}
          locale={lang}
          currentLocale={currentLocale}
        />
      ))}
    </div>
  );
}
