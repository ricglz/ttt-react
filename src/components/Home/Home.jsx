import React from 'react';
import { FormattedMessage } from 'react-intl';
import HomeButton from './HomeButton';

function Home() {
  return (
    <div className="container text-center">
      <div className="row">
        <div className="col">
          <FormattedMessage id="homePage.title" default="Home Page">
            {txt => <h1>{txt}</h1>}
          </FormattedMessage>
        </div>
      </div>
      <HomeButton text="shared.sp" url="/singleplayer" />
      <HomeButton text="shared.mp" url="/multiplayer" />
      <HomeButton staticText text="Online" url="/online" />
      <HomeButton text="shared.tutorial" url="/tutorial" />
      <HomeButton staticText text="Contributors" url="/contributors" />
    </div>
  );
}

export default Home;
