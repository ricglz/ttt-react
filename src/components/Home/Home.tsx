import React from 'react';
import { FormattedHeader } from '../Layout/FormattedText';
import HomeButton from './HomeButton';

function Home() {
  return (
    <div className="container text-center">
      <div className="row">
        <div className="col">
          <FormattedHeader locale="homePage.title" defaultMessage="Home Page" />
        </div>
      </div>
      <HomeButton text="shared.sp" url="/singleplayer" />
      <HomeButton text="shared.mp" url="/multiplayer" />
      <HomeButton staticText text="Online" url="/online" />
      <HomeButton text="shared.tutorial" url="/tutorial" />
      <HomeButton staticText text="Contributors" url="/contributors" />
    </div>
  );
}

export default Home;
