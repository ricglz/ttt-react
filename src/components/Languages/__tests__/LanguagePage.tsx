import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../../Home/Home';
import LanguagePage from '../LanguagePage';
import ContextsProvider from '../../../ContextsProvider';

function Component() {
  return (
    <ContextsProvider
      ChildrenComponent={Home}
      RouterComponent={MemoryRouter}
    >
      <LanguagePage />
    </ContextsProvider>
  );
}

it('renders without crashing', () => {
  const component = <Component />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
