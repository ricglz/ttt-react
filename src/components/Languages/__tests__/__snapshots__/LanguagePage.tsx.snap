// Vitest Snapshot v1

exports[`renders without crashing 1`] = `
<div
  className="text-left"
  dir="ltr"
>
  <div
    className="container text-center"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h1>
          Home Page
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/singleplayer"
        onClick={[Function]}
      >
        Single Player
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/multiplayer"
        onClick={[Function]}
      >
        Local Multiplayer
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/online"
        onClick={[Function]}
      >
        <span>
          Online
        </span>
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/tutorial"
        onClick={[Function]}
      >
        How to play
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/contributors"
        onClick={[Function]}
      >
        <span>
          Contributors
        </span>
      </a>
    </div>
  </div>
  <div
    className="row justify-content-center mt-5"
  >
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button current-locale"
        onClick={[Function]}
        type="button"
      >
        en
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        es
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        fr
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        pt
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        de
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        it
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        hi
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        mr
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        ko
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        ja
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        da
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        sr
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        id
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        uk
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        ru
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        sv
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        zh
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        zhCN
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        ar
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        ca
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        cs
      </button>
    </div>
    <div
      className="col-4 col-lg-4 col-xl-4 border-right border-top"
    >
      <button
        className="btn w-100 locale-button"
        onClick={[Function]}
        type="button"
      >
        tr
      </button>
    </div>
  </div>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      {
        "bottom": 16,
        "left": 16,
        "pointerEvents": "none",
        "position": "fixed",
        "right": 16,
        "top": 16,
        "zIndex": 9999,
      }
    }
  />
</div>
`;
